x-pack/plugins/enterprise_search/public/applications/enterprise_search_overview/components/elasticsearch_guide/elasticsearch_guide.tsx

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState } from 'react';
+
+import queryString from 'query-string';
+
+import {
+  EuiPageTemplate,
+  EuiText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiSteps,
+  EuiSelect,
+  EuiLink,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { ElasticsearchClientInstructions } from '../elasticsearch_client_instructions';
+import { ElasticsearchCloudId } from '../elasticsearch_cloud_id';
+import { ElasticsearchResources } from '../elasticsearch_resources';
+
+// Replace FormattedMessage with i18n strings
+
+export const ElasticsearchGuide: React.FC = () => {
+  const languages = [
+    { value: 'dotnet', text: '.Net' },
+    { value: 'go', text: 'Go' },
+    { value: 'java', text: 'Java' },
+    { value: 'javascript', text: 'JavaScript' },
+    { value: 'php', text: 'PHP' },
+    { value: 'python', text: 'Python' },
+    { value: 'ruby', text: 'Ruby' },
+    { value: 'rust', text: 'Rust' },
+  ];
+
+  const client = queryString.parse(window.location.search).client;
+  const languageExists = languages.some((language) => language.value === client);
+  const [selectedLanguage, setSelectedLanguage] = useState(
+    languageExists ? (client as string) : 'java'
```

Comment:
I suggest moving the string cast to line 44

Reply:
Yeah, nice catch, useState should usually be higher. But in this case, it seems necessary, because I'm using the `languageExists` variable that I define above. Or did I misunderstand something?